velox/functions/prestosql/InPredicate.cpp

```diff
@@ -322,6 +323,13 @@ class InPredicate : public exec::VectorFunction {
                                   .argumentType("array(unknown)")
                                   .build());
     }
+    // logical type: Date
+    signatures.emplace_back(exec::FunctionSignatureBuilder()
```

Comment:
This doesn't look right.  Shouldn't 2nd argument be array(date)?

CC: @aditi-pandit

Reply:
Not an expert on the compiled expression, but when debugging thru `compileRewrittenExpression`, the `inputTypes` does match `(DATE, ARRAY<INTEGER>)`

Reply:
@mbasmanova , @ericyuliu : The In expression construction in Presto_cpp used the LogicalType of DATE causing this issue. Have a fix https://github.com/prestodb/presto/pull/20521

Reply:
@aditi-pandit Aditi, thank you for root-causing and rolling out a fix.